tests/test_preference_comparisons.py

```diff
@@ -8,22 +10,100 @@
 from imitation.util import util
 
 
-def test_trainer_no_crash():
-    venv = util.make_vec_env(
+@pytest.fixture
+def venv():
+    return util.make_vec_env(
         "CartPole-v1",
         n_envs=1,
     )
-    reward_net = reward_nets.BasicRewardNet(venv.observation_space, venv.action_space)
-    # verbose=1 suppresses SB3 logger configuration,
-    # which conflicts with imitation logging
-    agent = stable_baselines3.PPO(
-        "MlpPolicy", venv, verbose=1, n_epochs=1, batch_size=2, n_steps=10
+
+
+@pytest.fixture
+def reward_net(venv):
+    return reward_nets.BasicRewardNet(venv.observation_space, venv.action_space)
+
+
+@pytest.fixture
+def agent(venv):
+    return stable_baselines3.PPO(
+        "MlpPolicy", venv, n_epochs=1, batch_size=2, n_steps=10
     )
-    agent_trainer = trainer.AgentTrainer(agent, reward_net)
-    fragmenter = preference_comparisons.RandomFragmenter(
+
+
+@pytest.fixture
+def fragmenter():
+    return preference_comparisons.RandomFragmenter(
         fragment_length=2, num_pairs=2, seed=0, warning_threshold=0
     )
+
+
+@pytest.fixture
+def agent_trainer(agent, reward_net):
+    return trainer.AgentTrainer(agent, reward_net)
+
+
+def test_trainer_no_crash(agent_trainer, reward_net, fragmenter):
     main_trainer = preference_comparisons.PreferenceComparisons(
         agent_trainer, reward_net, sample_steps=10, fragmenter=fragmenter
     )
     main_trainer.train(2)
+
+
+def test_discount_rate_no_crash(agent_trainer, reward_net, fragmenter):
+    # also use a non-zero noise probability to ensure that doesn't cause errors
```

Comment:
```suggestion
    # also use a non-zero noise probability to check that doesn't cause errors
```
I initially parsed this as saying we need non-zero noise probability otherwise it'll error, but I think this is saying that we're also testing that?